### PR TITLE
Validations and specs

### PR DESCRIPTION
**Pull request by bengigo**
**Changes made on this branch:**

Add unit specs for all models' methods and validations.
Add the following validations 
- For the User model:
   - Name must not be blank.
   - PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
   - Title must not be blank.
   - Title must not exceed 250 characters.
   - CommentsCounter must be an integer greater than or equal to zero.
   - LikesCounter must be an integer greater than or equal to zero.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   belongs_to :post
```
Comment (rloterh) [inline app/models/like.rb:4]: - [x] Validation are missing for `CommentsCounter` and `LikesCounter`. Kindly ensure to add validation(s) for the `CommentsCounter` and `LikesCounter` such that they're an integer greater than or equal to zero.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Bengi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good Rspec unit test coverage

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- Kindly include instructions in your ReadMe on how to run the test cases for your project, under the _Getting Started_ section.
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (bengigo) [reply app/models/like.rb:4]: @rloterh thank you for your review. The reason they are missing is they were not required for the project and I only created the required ones. But I added them now according to your review. 

Screenshot 2022-10-28 at 15 57 37

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello    @bengigo     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x] Your project is professional 

- [x]  Your repo is professional 

- [x] No Linter Errors






Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (rloterh) [reply app/models/like.rb:4]: @bengigo 
This was addressed in the Post model. I apologize for missing this step. Cheers, and happy coding! 😃

**Pull request merged**